### PR TITLE
improves e2e test of create scheduler

### PR DESCRIPTION
**Pull request by caiordjesus**
### Why?
We want to be sure that the OnError routine of create scheduler is working properly

### How?
Creating a new e2e test for "create scheduler", where the operation fails

### CODE REVIEW

File: e2e/suites/management/create_scheduler_test.go [hunk 1]
```diff
@@ -28,80 +28,165 @@ import (
...
+				listOperationsRequest := &maestrov1.ListOperationsRequest{}
+				listOperationsResponse := &maestrov1.ListOperationsResponse{}
+				err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/%s/operations", schedulerName), listOperationsRequest, listOperationsResponse)
+				assert.NoError(t, err)
+
+				if len(listOperationsResponse.FinishedOperations) == 0 {
+					return false
+				}
+
+				// TODO(gabrielcorado): we can use the operations constants here.
```
Comment (arthur29) [inline e2e/suites/management/create_scheduler_test.go:100]: Hey @caiordjesus WDYT to apply this comment and remove it?

File: e2e/suites/management/create_scheduler_test.go [hunk 2]
```diff
@@ -28,80 +28,165 @@ import (
...
+				assert.Equal(t, "create_scheduler", listOperationsResponse.FinishedOperations[0].DefinitionName)
+				assert.Equal(t, "error", listOperationsResponse.FinishedOperations[0].Status)
+				return true
+			}, 30*time.Second, time.Second)
+
+			// Check that scheduler does not exist since operation finished with error
+			getSchedulerRequest := &maestrov1.GetSchedulerRequest{}
+			getSchedulerResponse := &maestrov1.GetSchedulerResponse{}
+			err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/%s", schedulerName), getSchedulerRequest, getSchedulerResponse)
+			assert.Error(t, err)
```
Comment (henriqueoelze) [inline e2e/suites/management/create_scheduler_test.go:178]: Should we test the specific error to avoid false success?

**Review by arthur29 — CHANGES_REQUESTED** [review]

**Review by henriqueoelze — APPROVED** [review]
🚢

**Review by guilhermocc — APPROVED** [review]

**Review by arthur29 — APPROVED** [review]
LGTM

**Pull request merged**